review/review.py

```diff
@@ -0,0 +1,124 @@
+# pylint: disable=import-error
+''' Review XBlock '''
+
+import logging
+
+import pkg_resources
+
+from xblock.core import XBlock
+from xblock.fields import Integer, Scope, String
+from xblock.fragment import Fragment
+from xblockutils.resources import ResourceLoader
+
+from .configuration import SHOW_PROBLEMS, SHOW_VERTICAL
+from .get_review_ids import get_problems, get_vertical
+
+log = logging.getLogger(__name__)
+loader = ResourceLoader(__name__)
+
+
+# Make '_' a no-op so we can scrape strings. Using below function instead of
+# `django.utils.translation.ugettext_noop` because Django cannot be imported in this file
+def _(text):
+    return text
+
+
+@XBlock.needs('i18n')
+class ReviewXBlock(XBlock):
+    '''
+    The Review XBlock helps learners review concepts from a course by redisplaying
+    a handful of problems from the course with a fresh state that are ungraded and
+    have unlimited attempts.
+    '''
+    display_name = String(
+        display_name=_('Display Name'),
+        help=_('The display name for this component.'),
+        scope=Scope.settings,
+        default=_('Review'),
+    )
+
+    num_desired = Integer(
+        display_name=_('Number of desired review problems'),
+        help=_('Defines the number of problems the review module will display '
+               'to the learner. You do not need this if displaying a full unit.'),
+        default=5,
+        values={"min": 2},
+        scope=Scope.content
+    )
+
+    def resource_string(self, path):
+        '''Handy helper for getting resources from our kit.'''
+        data = pkg_resources.resource_string(__name__, path)
+        return data.decode('utf8')
+
+    def get_problem_html(self):
+        '''
+        Create the html for showing review problems by picking individual
+        problems. This calls the get_problems function which will return
+        self.num_desired urls to display in iFrames for the Review XBlock.
+        '''
+        # url_list elements have the form (url, correctness, attempts)
+        url_list = get_problems(self.num_desired, self.course_id)
+        if len(url_list) == self.num_desired:
```

Comment:
Nit: You may want to just check for a non-empty return value here - as the helper function returns an empty list if the desired number isn't satisfied.